test_package/conanfile.py

```diff
@@ -24,6 +24,11 @@ def test(self):
         if self.settings.os == "Windows":
             path = self.build_folder + os.sep \
                 + str(self.settings.build_type) + os.sep
-            self.run("%sexample.exe" % path)
+            if self.options["libmodbus"].shared:
+                runcwd = self.build_folder + os.sep \
```

Comment:
You can probably use `runcwd = os.path.join(self.build_folder, "bin")` here instead.